google/api/http.proto

```diff
@@ -346,6 +347,20 @@ message HttpRule {
   // message type.
   string body = 7;
 
+  // A mapping from HTTP headers to request fields. If
+  // a header has a case-insensitive match to a key in this map, the field in
+  // the request that matches the value will have its value set to the value
+  // in the header.
```

Comment:
to the value in the header => to the value of the HTTP header.

We need to specify the precise encoding for string, bytes, number, etc.